logstash-core/src/test/java/org/logstash/EventTest.java

```diff
@@ -453,4 +453,19 @@ public void metadataRootShouldBeValuefied() {
         assertEquals(list, Arrays.asList("hello"));
 
     }
+
+    @Test
+    public void removeMetadataField() {
```

Comment:
there are two implementations in code depending on whether we remove an item _from_ `@metadata` or the `@metadata` itself, so we should test both cases like we do in the ruby spec.

Reply:
agree. As noted in the description TODO, there are still a few missing test cases and I first wanted to make sure we agree on the approach before committing to all tests!

Reply:
added.